src/Features/CSharp/Portable/BraceCompletion/InterpolatedStringBraceCompletionService.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.BraceCompletion;
+using Microsoft.CodeAnalysis.CSharp.Extensions.ContextQuery;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.CSharp.BraceCompletion
+{
+    /// <summary>
+    /// Brace completion service for double quotes marking an interpolated string.
+    /// Note that the <see cref="StringLiteralBraceCompletionService"/> is used for
+    /// other double quote completions.
+    /// </summary>
+    [Export(LanguageNames.CSharp, typeof(IBraceCompletionService)), Shared]
+    internal class InterpolatedStringBraceCompletionService : AbstractBraceCompletionService
+    {
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public InterpolatedStringBraceCompletionService()
+        {
+        }
+
+        protected override char OpeningBrace => DoubleQuote.OpenCharacter;
+
+        protected override char ClosingBrace => DoubleQuote.CloseCharacter;
+
+        public override Task<bool> AllowOverTypeAsync(BraceCompletionContext context, CancellationToken cancellationToken)
+            => AllowOverTypeWithValidClosingTokenAsync(context, cancellationToken);
+
+        /// <summary>
+        /// Only return this service as valid when we're starting an interpolated string.
+        /// Otherwise double quotes should be completed using the <see cref="StringLiteralBraceCompletionService"/>
+        /// </summary>
+        public override async Task<bool> CanProvideBraceCompletionAsync(char brace, int openingPosition, Document document, CancellationToken cancellationToken)
+            => OpeningBrace == brace && await IsPositionInInterpolatedStringContextAsync(document, openingPosition, cancellationToken).ConfigureAwait(false);
+
+        protected override bool IsValidOpeningBraceToken(SyntaxToken leftToken)
+            => leftToken.IsKind(SyntaxKind.InterpolatedStringStartToken) || leftToken.IsKind(SyntaxKind.InterpolatedVerbatimStringStartToken);
+
+        protected override bool IsValidClosingBraceToken(SyntaxToken rightToken)
+            => rightToken.IsKind(SyntaxKind.InterpolatedStringEndToken);
+
+        protected override Task<bool> IsValidOpenBraceTokenAtPositionAsync(SyntaxToken token, int position, Document document, CancellationToken cancellationToken)
+        {
+            return Task.FromResult(IsValidOpeningBraceToken(token)
+                && token.Span.End - 1 == position);
```

Comment:
this read very weird.  consider either:

```suggestion
            return Task.FromResult(IsValidOpeningBraceToken(token) && token.Span.End - 1 == position);
```

or

```suggestion
            return Task.FromResult(
                IsValidOpeningBraceToken(token) && token.Span.End - 1 == position);
```